script.js

```diff
@@ -9,3 +9,169 @@ document.querySelectorAll('.hide-menu').forEach((element) => {
+  if (i % 2 === 0) {
+    portfolioFlex.classList.add('portfolio-flex-row');
+  } else {
+    portfolioFlex.classList.add('portfolio-flex-row-reverse');
+  }
+  portfolioImage.classList.add('portfolio-img');
+  portfolioRight.classList.add('portfolio-right');
+  portfolioHeading.classList.add('portfolio-heading');
+  portfolioList.classList.add('portfolio-list');
+  portfolioText.classList.add('portfolio-text');
+  portfolioLanguages.classList.add('portfolio-languages');
+
+  portfolioHeading.textContent = portfolioObjects[object].heading;
+  portfolioList.innerHTML = portfolioObjects[object].subheading;
+  portfolioImage.src = portfolioObjects[object].image;
+  portfolioText.textContent = portfolioObjects[object].shortDescription;
+  portfolioLanguages.innerHTML = portfolioObjects[object].languages;
+  portfolioButton.textContent = portfolioObjects[object].seeProjectButton;
+  portfolioButton.addEventListener('click', () => {
+    document.querySelector('.header').classList.remove('header-fixed');
+
+    const popupDiv = document.createElement('div');
+    const divContainer = document.createElement('div');
+    const topBar = document.createElement('div');
+    const h2 = document.createElement('h2');
+    const close = document.createElement('span');
+    const subheading = document.createElement('ul');
+    const image = document.createElement('img');
+    const bottomFlex = document.createElement('div');
+    const rightBottomFlex = document.createElement('div');
+    const description = document.createElement('p');
+    const buttons = document.createElement('div');
+    const languages = document.createElement('ul');
+    const line = document.createElement('hr');
+    const source = document.createElement('button');
+    const liveVersion = document.createElement('button');
+
+    popupDiv.classList.add('popupdiv');
+    divContainer.classList.add('div-container');
+    topBar.classList.add('portfolio-top-bar');
+    h2.classList.add('portfolio-h2');
+    subheading.classList.add('subheading');
+    image.classList.add('popup-image');
+    bottomFlex.classList.add('popup-bottom-flex');
+    rightBottomFlex.classList.add('rightbottomflex');
+    description.classList.add('description');
+    buttons.classList.add('popup-buttons');
+    languages.classList.add('languages');
+    line.classList.add('line');
+    liveVersion.classList.add('liveVersion');
+
+    h2.textContent = portfolioObjects[object].heading;
+    subheading.innerHTML = portfolioObjects[object].subheading;
+    image.src = portfolioObjects[object].image;
+    description.textContent = portfolioObjects[object].description;
+    languages.innerHTML = portfolioObjects[object].languages;
+    source.href = portfolioObjects[object].source;
+    source.textContent = 'Source';
+    liveVersion.href = portfolioObjects[object].liveVersion;
+    liveVersion.textContent = 'Live Site';
+    close.textContent = 'x';
+    close.classList.add('close');
```

Comment:
This is great but alternatively, you could use a function that returns a template of HTML that accepts a portfolio object and populates the template then you can set the object with `innerHTML`.
Funciton would look something like this: (I didn't fix the formatting as I copied most of it)

```js
function (portofolioObject) => {
`<h2 class="portfolio-heading">Tonic</h2>
          <ul>
            <li class="portfolio-list-first">${portofolioObject.title}</li>
            <li class="portfolio-list-dot"></li>
            <li class="portfolio-list-secondary">${portofolioObject.example}</li>
            <li class="portfolio-list-dot"></li>
            <li class="portfolio-list-secondary">${portofolioObject.year}</li>
          </ul>
          <p class="portfolio-text">
            ${portofolioObject.example}
          </p>
          <ul>
 `
 }
 ```

Reply:
Yeah, this looks alot easier lol. Do you think i should change it? It doesn't seem like there's anything wrong with it as it is

Reply:
It's fine the way it is 👍